### PR TITLE
ks-linkfix-202107

### PR DESCRIPTION
**Pull request by v-kents**
This PR is the result of running a Link Repair script on the included content. This script is primarily being run to cleanup links to function correctly in Air Gapped Clouds (AGC). Here's a breakdown of what the script that generated the changes in this PR fixes:

- docs.microsoft.com Fully Qualified Domain Name (FQDN) Links to either Relative (if in same Repo/DocSet/ContentSet) and Site Relative if not
- azure.microsoft.com/documentation/articles that reflect to Docs to either Relative (if in same Repo/DocSet/ContentSet) and Site Relative if not
- All known flavors of MSDN/TechNet domains (we know of 8 or 9 now) that reflect to Docs to either Relative (if in same Repo/DocSet/ContentSet) and Site Relative if not
  - This also includes cleanup of the appended query string for MSDN (redirectedfrom=MSDN).
- Fixes articles that are redirected in `.openpublishing.redirection.json` and updates the link to the current final location (this has a lot of customer usability fixes).

Please note the [Contributor Guide: Links](https://review.docs.microsoft.com/en-us/help/contribute/links-how-to?branch=master) has been updated with the following link-type preference order:

```
By order of preference, links hosted on Docs should be Relative if they are in the same repository and docset. If they are in a different docset, even if in the same repository, they should be Site Relative. Links to content hosted on Docs shouldn't use a complete URL, otherwise known as Fully Qualified Domain Names (FQDN). Using a complete URL from Docs to other content on Docs will cause that link to be non-functional in air-gap environments.
```

### CODE REVIEW

File: docs/architecture/porting-existing-aspnet-apps/deployment-scenarios.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ Consider the common scenario of a large web app that currently is hosted on IIS
...
+Fortunately, IIS is a very feature-rich web server, and two features it has had for some time are its [URL Rewrite module and Application Request Routing](/iis/extensions/url-rewrite-module/reverse-proxy-with-url-rewrite-v2-and-application-request-routing). Using these features, IIS can act as a [reverse proxy](/iis/extensions/url-rewrite-module/reverse-proxy-with-url-rewrite-v2-and-application-request-routing), routing client requests to the appropriate back-end web app. To configure IIS as a reverse proxy, check the **Enable proxy** checkbox in the Application Request Routing feature, then add a URL Rewrite rule like this one:
```
Comment (gewarren) [inline docs/architecture/porting-existing-aspnet-apps/deployment-scenarios.md:20]: These two links are the same - wonder if that's a mistake.

**Review by gewarren — APPROVED** [review]

File: docs/architecture/porting-existing-aspnet-apps/deployment-scenarios.md (hunk 1 above)
Comment (v-kents) [reply docs/architecture/porting-existing-aspnet-apps/deployment-scenarios.md:20]: @gewarren Probably one of them used to be somewhere else, but was redirected. The script replaces redirections with the target URL. How would you prefer it be fixed? Remove the second link, or replace with something else?

Comment (gewarren) [inline docs/architecture/porting-existing-aspnet-apps/deployment-scenarios.md]:
```suggestion
Fortunately, IIS is a feature-rich web server, and two features it has are [URL Rewrite module](/iis/extensions/url-rewrite-module/url-rewrite-module-video-walkthrough) and [Application Request Routing](/iis/extensions/planning-for-arr/application-request-routing-version-2-overview). Using these features, IIS can act as a [reverse proxy](/iis/extensions/url-rewrite-module/reverse-proxy-with-url-rewrite-v2-and-application-request-routing), routing client requests to the appropriate back-end web app. To configure IIS as a reverse proxy, check the **Enable proxy** checkbox in the Application Request Routing feature, then add a URL Rewrite rule like this one:
```

Comment (gewarren) [reply docs/architecture/porting-existing-aspnet-apps/deployment-scenarios.md]: How about this?

Comment (v-kents) [reply docs/architecture/porting-existing-aspnet-apps/deployment-scenarios.md]: works for me, thank you!

**Comment by v-kents** [comment]
#sign-off (bulk updates to links - and that suggestion was applied, despite it not looking like it above)

**Pull request merged**